packages/web/src/features/translation/TranslationView.tsx

```diff
@@ -32,34 +30,58 @@ export default function TranslationView() {
   >([]);
   const queryClient = useQueryClient();
   const glossMutation = useMutation({
-    mutationFn: (variables: { wordId: string; gloss: string }) =>
+    mutationFn: (variables: { wordId: string; gloss?: string }) =>
       apiClient.words.updateGloss(variables.wordId, language, variables.gloss),
-    onMutate: ({ wordId }) => {
+    onMutate: async ({ wordId, gloss }) => {
       const requestId = Math.floor(Math.random() * 1000000);
       setGlossRequests((requests) => [...requests, { wordId, requestId }]);
-      return { requestId };
+
+      const queryKey = ['verse-glosses', language, params.verseId];
+      await queryClient.cancelQueries({ queryKey });
+      const previousGlosses = queryClient.getQueryData(queryKey);
+      queryClient.setQueryData<GetVerseGlossesResponseBody>(queryKey, (old) => {
```

Comment:
This mutation to update glosses is a little more complex now. When we fire the mutation we optimistically update the cache with the new gloss.